docs/docs/explanations/configuration-best-practices.mdx

```diff
@@ -0,0 +1,63 @@
+---
+title: Configuration best practices
+id: config-best-practices
+description: A guide to best practices for configuring Nebari
+---
+
+This document highlights and consolidates configuration best practices that are introduced throughout the Getting Started, How To's, and Advanced Settings sections of the documentation.
+
+### Naming conventions
+
+When Nebari creates a new cloud resource on each one of the cloud providers, it will have to provide properties such as a name, project id, buckets, etc. for these resources.
+As Nebari mainly uses the project name from config to populate those attributes, we must ensure that for such cases, the resource attribute/name meets the bound conventions for the chosen cloud provider.
+
+To avoid any burden on the user, on keeping track of each new naming rule for all infrastructure related resources that Nebari creates, we have automated the process to ensure that all resources are named in a consistent manner.
+The necessary restrictions for each cloud provider are presented below:
+
+- Letters from A to Z (upper and lower case) and numbers
+- Maximum accepted length of the name string is **16** characters
+- If using `AWS`: names should not start with the string "aws"
+- If using `Azure`: names should not contain "-"
+
+
+### Domain Format
+
+### Omitting sensitive values
+
+If you wish to avoid storing secrets directly in the config yaml file you can instead set the values in environment variables. This substitution is triggered by setting config
+values to `QHUB_SECRET_` followed by the environment variable name.
+
+For example, you could set the environment variables `github_client_id` and `github_client_key` and write the
+following in your config file:
+
+```yaml
+security:
+  authentication:
+    type: GitHub
+    config:
+      client_id: QHUB_SECRET_github_client_id
+      client_secret: QHUB_SECRET_github_client_key
+```
+
+### Nebari version assertion
+
+All `qhub-config.yaml` files must contain a `qhub_version` field displaying the version of Nebari which it’s intended to be deployed with.
+
+Typically, you can upgrade the `qhub-config.yaml` file itself by manually adding the intended version to the file and validating that the version matches the version of Nebari you are deploying with by calling `qhub --version`.
+
+If available, you may also update the configuration using the `qhub upgrade` command. This will update image numbers, plus updating `qhub_version` to match the installed version of qhub, as well as any other bespoke changes required.
```

Comment:
```suggestion
If available, you may also update the configuration using the `nebari upgrade` command. This will update image numbers, plus update `nebari_version` to match the installed version of Nebari, as well as any other bespoke changes required.
```